fixes and additions

eyupsayin opened the pull request:
* "stdint.h" and "NUC1311.h" libraries cannot be accessed because they are not called in the files. It is not reflected in the error counter when compiling in Keil, but it appears as an error in the file. Fixed.
* "Sign Compare", "Unused Parameter", "auto logical bitwise compare" and "unused but set parameter" warnings have been fixed.
* CAN receiver masking setting function is missing. Added.

eyupsayin committed:
Uncalled libraries in files

"stdint.h" and "NUC1311.h" libraries cannot be accessed because they are not called in the files. It is not reflected in the error counter when compiling in Keil, but it appears as an error in the file. Fixed.

eyupsayin committed:
Warnings fixed

"Sign Compare", "Unused Parameter", "auto logical bitwise compare" and "unused but set parameter" warnings have been fixed.

eyupsayin committed:
CAN_SetRxMsgAndMsk() func. added.

CAN receiver masking setting function is missing. Added.

wschang0 commented:
Thanks for your feedback. 
However, we double check the files and it looks fine.
The "stdint.h" should be included in "core_cm0.h" which will be include by "NUC1311.h"
In our BSP, include "NUC1311.h" will also include all driver headers.

eyupsayin commented:
Hi,
When the "NUC1311.h" file is not added to the files added as a commit, the following image is generated when compiled with the Keil Compiler v6 c99.

![pwmHeaderScSh](https://github.com/user-attachments/assets/76f49a13-e8e4-4d43-82bc-ae99e1b4e73e)

eyupsayin committed:
Update pwm.h

To improve understandability, the PWM_SET_DUTY macro has been added, which is equal to the PWM_SET_CMR macro, and the PWM_SET_PERIOD macro has been added, which is equal to the PWM_SET_CNR macro.
